internal/controllers/user_integrations_controller.go

```diff
@@ -1933,6 +1950,49 @@ func (udc *UserDevicesController) registerDeviceTesla(c *fiber.Ctx, logger *zero
 	return c.SendStatus(fiber.StatusNoContent)
 }
 
+func (udc *UserDevicesController) getTeslaAuthFromCache(ctx context.Context, userID string) (*RegisterDeviceIntegrationRequest, error) {
+	user, err := udc.usersClient.GetUser(ctx, &pb.GetUserRequest{Id: userID})
+	if err != nil {
+		return nil, errors.Wrap(err, "could not fetch user information")
+	}
+
+	if user.EthereumAddress == nil {
+		return nil, fmt.Errorf("missing wallet address for user")
+	}
+
+	cacheKey := fmt.Sprintf(TeslaFleetAuthCacheKey, *user.EthereumAddress)
+	encTeslaAuth, err := udc.redisCache.Get(ctx, cacheKey).Result()
+	if err != nil {
+		udc.log.Err(err).Str("Cache Key", cacheKey).Msg("Error occurred retrieving tesla auth from cache using the key")
+		return nil, errors.Wrap(err, "an error occurred completing tesla device registration")
```

Comment:
Can we be more specific?

I'm a `fmt.Errorf` guy, personally, but `errors.Wrap` is good too.